File: OPENAPI-REST-API/openapi-client/php-dt/src/App/DTO/ServerServiceInfo.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

class ServerServiceInfo
{
    /**
     * The ID of the server.
     */
    #[DTA\Data(field: "server_id", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_id = null;

    /**
     * The hostname of the server.
     */
    #[DTA\Data(field: "server_hostname", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_hostname = null;

    /**
     * The customer ID associated with the server.
     */
    #[DTA\Data(field: "server_custid", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_custid = null;

    /**
     * The type of the server.
     */
    #[DTA\Data(field: "server_type", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_type = null;

    /**
     * The currency used for billing.
     */
    #[DTA\Data(field: "server_currency", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_currency = null;

    /**
     * The date when the server was ordered.
     */
    #[DTA\Data(field: "server_order_date", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_order_date = null;

    /**
     * The invoice number for the server.
     */
    #[DTA\Data(field: "server_invoice", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_invoice = null;

    /**
     * The coupon associated with the server.
     */
    #[DTA\Data(field: "server_coupon", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_coupon = null;

    /**
     * The status of the server.
     */
    #[DTA\Data(field: "server_status", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_status = null;

    /**
     * The root of the server.
     */
    #[DTA\Data(field: "server_root", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_root = null;

    /**
     * The dedicated tag of the server.
     */
    #[DTA\Data(field: "server_dedicated_tag", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_tag = null;

    /**
     * The custom tag of the server.
     */
    #[DTA\Data(field: "server_custom_tag", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_custom_tag = null;

    /**
     * Comments related to the server.
     */
    #[DTA\Data(field: "server_comment", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_comment = null;

    /**
     * The initial billing amount for the server.
     */
    #[DTA\Data(field: "server_initial_bill", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_initial_bill = null;

    /**
     * The hardware information of the server.
     */
    #[DTA\Data(field: "server_hardware", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_hardware = null;

    /**
     * The number of IPs associated with the server.
     */
    #[DTA\Data(field: "server_ips", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_ips = null;

    /**
     * The monthly billing amount for the server.
     */
    #[DTA\Data(field: "server_monthly_bill", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_monthly_bill = null;

    /**
     * The setup status of the server.
     */
    #[DTA\Data(field: "server_setup", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_setup = null;

    /**
     * Discount information for the server.
     */
    #[DTA\Data(field: "server_discount", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "mixed"])]
    public mixed|null $server_discount = null;

    /**
     * The reputation of the server.
     */
    #[DTA\Data(field: "server_rep", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_rep = null;

    /**
     * The date related to the server.
     */
    #[DTA\Data(field: "server_date", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_date = null;

    /**
     * The total cost of the server.
     */
    #[DTA\Data(field: "server_total_cost", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_total_cost = null;

    /**
     * The location of the server.
     */
    #[DTA\Data(field: "server_location", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "mixed"])]
    public mixed|null $server_location = null;

    /**
     * The ordered hardware for the server.
     */
    #[DTA\Data(field: "server_hardware_ordered", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_hardware_ordered = null;

    /**
     * The billed amount for the server.
     */
    #[DTA\Data(field: "server_billed", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_billed = null;

    /**
     * Indicates whether a welcome email was sent.
     */
    #[DTA\Data(field: "server_welcome_email", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_welcome_email = null;

    /**
     * The number of dedicated CPUs for the server.
     */
    #[DTA\Data(field: "server_dedicated_cpu", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_cpu = null;

    /**
     * The amount of dedicated memory for the server.
     */
    #[DTA\Data(field: "server_dedicated_memory", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_memory = null;

    /**
     * The size of the first dedicated hard drive.
     */
    #[DTA\Data(field: "server_dedicated_hd1", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_hd1 = null;

    /**
     * The size of the second dedicated hard drive.
     */
    #[DTA\Data(field: "server_dedicated_hd2", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "mixed"])]
    public mixed|null $server_dedicated_hd2 = null;

    /**
     * The bandwidth of the server.
     */
    #[DTA\Data(field: "server_dedicated_bandwidth", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_bandwidth = null;

    /**
     * The number of dedicated IPs for the server.
     */
    #[DTA\Data(field: "server_dedicated_ips", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_ips = null;

    /**
     * The operating system of the server.
     */
    #[DTA\Data(field: "server_dedicated_os", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_os = null;

    /**
     * The control panel of the server.
     */
    #[DTA\Data(field: "server_dedicated_cp", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "mixed"])]
    public mixed|null $server_dedicated_cp = null;

    /**
     * The RAID configuration of the server.
     */
    #[DTA\Data(field: "server_dedicated_raid", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_dedicated_raid = null;

    /**
     * Additional information about the server.
     */
    #[DTA\Data(field: "server_extra", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $server_extra = null;

}

File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/ServerServiceInfo.scala
package io.swagger.server.model


/**
 * @param server_id The ID of the server. for example: ''16058''
 * @param server_hostname The hostname of the server. for example: ''myserver.host.com''
 * @param server_custid The customer ID associated with the server. for example: ''771282''
 * @param server_type The type of the server. for example: ''600''
 * @param server_currency The currency used for billing. for example: ''USD''
 * @param server_order_date The date when the server was ordered. for example: ''2020-05-08T17:22:36.000Z''
 * @param server_invoice The invoice number for the server. for example: ''18738142''
 * @param server_coupon The coupon associated with the server. for example: ''0''
 * @param server_status The status of the server. for example: ''active''
 * @param server_root The root of the server.
 * @param server_dedicated_tag The dedicated tag of the server. for example: ''0''
 * @param server_custom_tag The custom tag of the server.
 * @param server_comment Comments related to the server.
 * @param server_initial_bill The initial billing amount for the server. for example: ''0''
 * @param server_hardware The hardware information of the server. for example: ''0''
 * @param server_ips The number of IPs associated with the server. for example: ''0''
 * @param server_monthly_bill The monthly billing amount for the server. for example: ''0''
 * @param server_setup The setup status of the server. for example: ''0''
 * @param server_discount Discount information for the server.
 * @param server_rep The reputation of the server. for example: ''0''
 * @param server_date The date related to the server. for example: ''1588972956''
 * @param server_total_cost The total cost of the server. for example: ''230''
 * @param server_location The location of the server.
 * @param server_hardware_ordered The ordered hardware for the server. for example: ''0''
 * @param server_billed The billed amount for the server. for example: ''0''
 * @param server_welcome_email Indicates whether a welcome email was sent. for example: ''1''
 * @param server_dedicated_cpu The number of dedicated CPUs for the server. for example: ''38''
 * @param server_dedicated_memory The amount of dedicated memory for the server. for example: ''22''
 * @param server_dedicated_hd1 The size of the first dedicated hard drive. for example: ''20''
 * @param server_dedicated_hd2 The size of the second dedicated hard drive.
 * @param server_dedicated_bandwidth The bandwidth of the server. for example: ''3''
 * @param server_dedicated_ips The number of dedicated IPs for the server. for example: ''5''
 * @param server_dedicated_os The operating system of the server. for example: ''30''
 * @param server_dedicated_cp The control panel of the server.
 * @param server_dedicated_raid The RAID configuration of the server. for example: ''0''
 * @param server_extra Additional information about the server. for example: ''[]''
 */
case class ServerServiceInfo (
  server_id: Option[String],
  server_hostname: Option[String],
  server_custid: Option[String],
  server_type: Option[String],
  server_currency: Option[String],
  server_order_date: Option[String],
  server_invoice: Option[String],
  server_coupon: Option[String],
  server_status: Option[String],
  server_root: Option[String],
  server_dedicated_tag: Option[String],
  server_custom_tag: Option[String],
  server_comment: Option[String],
  server_initial_bill: Option[String],
  server_hardware: Option[String],
  server_ips: Option[String],
  server_monthly_bill: Option[String],
  server_setup: Option[String],
  server_discount: Option[String],
  server_rep: Option[String],
  server_date: Option[String],
  server_total_cost: Option[String],
  server_location: Option[String],
  server_hardware_ordered: Option[String],
  server_billed: Option[String],
  server_welcome_email: Option[String],
  server_dedicated_cpu: Option[String],
  server_dedicated_memory: Option[String],
  server_dedicated_hd1: Option[String],
  server_dedicated_hd2: Option[String],
  server_dedicated_bandwidth: Option[String],
  server_dedicated_ips: Option[String],
  server_dedicated_os: Option[String],
  server_dedicated_cp: Option[String],
  server_dedicated_raid: Option[String],
  server_extra: Option[String]
)

